tests/integration/internal/receivermock/receiver_mock.go

```diff
@@ -122,6 +122,41 @@ func (client *ReceiverMockClient) GetMetricsSamples(
 	return metricsSamples, nil
 }
 
+type LogsCountResponse struct {
+	Count uint
+}
+
+func (client *ReceiverMockClient) GetLogsCount(t *testing.T, metadataFilters MetadataFilters) (uint, error) {
+	path, err := url.Parse("logs/count")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	queryParams := url.Values{}
+	for key, value := range metadataFilters {
+		queryParams.Set(key, value)
+	}
+
+	url := client.baseUrl.ResolveReference(path)
+	url.RawQuery = queryParams.Encode()
+
+	statusCode, body := http_helper.HttpGet(
+		t,
+		url.String(),
+		&client.tlsConfig,
+	)
+	if statusCode != 200 {
+		return 0, fmt.Errorf("received status code %d in response to receiver request", statusCode)
+	}
+
+	var response LogsCountResponse
+	err = json.Unmarshal([]byte(body), &response)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return response.Count, nil
```

Comment:
Why `t.Fatal`? Isn't that just an error which you can return from this func?

Reply:
Same reason we Fatal if url parsing fails, it almost always implies a bug in the code, and shouldn't just be retried. It's simpler this way because the caller doesn't need to inspect the error and decide whether they should Fatal themselves or continue.

Reply:
Got it. Kinda makes sense but then why do we use the helper that does the http get which  requires `t` (on which it can set the test as failed)? This might be a transient error that we don't want to fail our whole testcase on, right?

Reply:
Good point, I should switch to the helper that doesn't fail tests and handle the error explicitly. I like the terratest helpers in general because the logging they provide is useful for debugging.